src/Database/Query.php

```diff
@@ -237,15 +242,40 @@ public function getConnection(): Connection
      * This method can be overridden in query subclasses to decorate behavior
      * around query execution.
      *
-     * @return \Cake\Database\StatementInterface
+     * @return \Cake\Database\Statement
      */
-    public function execute(): StatementInterface
+    public function execute(): Statement
     {
-        $statement = $this->_connection->run($this);
-        $this->_iterator = $this->_decorateStatement($statement);
+        $this->_statement = $this->_connection->run($this);
         $this->_dirty = false;
 
-        return $this->_iterator;
+        return $this->_statement;
+    }
+
+    /**
+     * Executes query and returns entire result set.
```

Comment:
This reads like the output will be a ResultSet.
```suggestion
     * Executes query and returns the decorated results.
```